### PR TITLE
Make sure oneshot services restart upon failure

### PR DESCRIPTION
**Pull request by saschat**
The password fetcher is failing. I am not sure why it suddenly happens (it worked before) but since we have no longer root access to the machine it is hard to investigate.

For now, lets make sure the service restarts on failure and check if the problem persists.

### CODE REVIEW

**Comment by saschat** [comment]
Given that our marketplace image is not working I will merge and test at once (I can only test this on AWS for now). @alevy feel free to still review if you want.

File: cachesight/default.nix [hunk 1]
```diff
@@ -161,6 +163,7 @@ in {
       path = [pkgs.curl];
       serviceConfig = {
         Type = "oneshot";
+        Restart = "on-failure";
```
Comment (alevy) [inline cachesight/default.nix:166]: I wonder if maybe three service starts sometimes after the network is to, but before it gets an address from dhcp

Comment (saschat) [reply cachesight/default.nix:166]: You are right. It seems requiring the `network.target` is pretty meaningless. I wounder why I see that everywhere then...

Comment (alevy) [reply cachesight/default.nix:166]: `network.target` is probably the right thing for servers that _listen_ (e.g. on all interfaces), which is more common

**Pull request merged**